app/controllers/experiments_controller.rb

```diff
@@ -369,11 +371,53 @@ def create
         replication_level: [:optional, :security_default, :integer, :positive],
         execution_time_constraint: [:optional, :security_default, :integer, :positive],
         parameter_constraints: [:optional, :security_json],
-        type: [:optional, :security_default]
+        type: [:optional, :security_default],
+        workers_scaling: [:optional] # TODO boolean validator
+        #workers_scaling_params: [:optional, :json_or_hash] TODO uncomment? SCAL-757: (..) json_or_hash validator bug
     )
+
+    # Params validation
     type = params[:type] || 'experiment'
-    raise ValidationError.new('type', type, 'Not a correct experiment type') unless CONSTRUCTORS.has_key? type
-    send(CONSTRUCTORS[type])
+    Utils::raise_error_unless_has_key(CONSTRUCTORS, type, "Not a correct experiment type: #{type}")
+    workers_scaling_params = nil
+    if params[:workers_scaling]
+      message_prefix = 'Missing workers scaling parameter'
+      Utils::raise_error_unless_has_key(params, :workers_scaling_params, message_prefix.pluralize)
+      workers_scaling_params = Utils::parse_json_if_string(params[:workers_scaling_params]).symbolize_keys
+      [:name, :allowed_infrastructures, :time_limit].each do |param|
+        Utils::raise_error_unless_has_key(workers_scaling_params, param, "#{message_prefix} #{param}",
+                                          'workers_scaling_params')
+      end
+      # TODO more precise validation
+    end
+    # Create experiment
+    experiment = send(CONSTRUCTORS[type])
+
+    # Start workers scaling
+    if params[:workers_scaling]
+      allowed_infrastructures = workers_scaling_params[:allowed_infrastructures].map do |record|
+        # TODO Introduce infrastructure id
+        {infrastructure: {name: record['name'].to_sym, params: record['params'].symbolize_keys}, limit: record['limit']}
+      end
+      # TODO proper parsing
+      allowed_infrastructures.each do |inf|
```

Comment:
@mwrona - tzn. jeśli w rekordzie "subinfrastructure" zapiszemy credentials_id jako string?
Teraz, kiedy where przyjmuje jakikolwiek klucz z *_id to automatycznie konwertuje wartość na BSON::ObjectId jeśli jest to string. query_simulation_manager_records używa wew. where, więc kiedy podamy w paramsach jako string to zadziała.